api/v1alpha2/terraform_types.go

```diff
@@ -766,6 +776,15 @@ func TerraformStateLocked(terraform *Terraform, lockID, message string) *Terrafo
 	return terraform
 }
 
+// TerraformStateLockReleased clears any state lock condition and resets lock tracking fields.
+func TerraformStateLockReleased(terraform *Terraform) *Terraform {
+	conditions.Delete(terraform, ConditionTypeStateLocked)
+	terraform.Status.Lock.Pending = ""
+	terraform.Status.Lock.LastApplied = ""
+
+	return terraform
```

Comment:
nit: Return value is never used, so maybe function should return anything? Or better yet, make it a method, since that IMO makes more sense when you only need to mutate the object.